contracts/interfaces/IP2PPairStaking.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.10;
+
+import "../dependencies/openzeppelin/contracts/IERC20.sol";
+
+interface IP2PPairStaking {
+    enum StakingType {
+        BAYCStaking,
+        MAYCStaking,
+        BAYCPairStaking,
+        MAYCPairStaking
+    }
+
+    struct ListingOrder {
+        StakingType stakingType;
+        address offerer;
+        address token;
+        uint32 tokenId;
+        uint32 share;
+        uint256 startTime;
+        uint256 endTime;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    struct MatchedOrder {
+        StakingType stakingType;
+        address apeToken;
+        address apeOfferer;
+        uint32 apeTokenId;
+        uint32 apeShare;
+        address bakcOfferer;
+        uint32 bakcTokenId;
+        uint32 bakcShare;
+        address apeCoinOfferer;
+        uint32 apeCoinShare;
+        uint256 apePrincipleAmount;
```

Comment:
we can use uint128 here

Reply:
apePrincipleAmount will consume a single slot no matter it is uint128 or uint256